include/mbedtls/sha1.h

```diff
@@ -131,11 +137,14 @@ int mbedtls_sha1_starts_ret( mbedtls_sha1_context *ctx );
  *                 constitutes a security risk. We recommend considering
  *                 stronger message digests instead.
  *
- * \param ctx      The SHA-1 context.
+ * \param ctx      The SHA-1 context. This must be initialized
+ *                 and have a hash operation started.
```

Comment:
It's desirable that a hash operations has started, but is it strictly required?

For example, I know TLS itself calls `mbedtls_md_process()` is on an initialized context without calling `mbedtls_md_starts()` first (intentionally, not a bug). This may trickle down into the SHA modules as `update()` calls that don't have `start()` called first.

I had a look at MD's implementation and it looks like this calls `mbedtls_internal_sha1_process()`, not the `mbedtls_sha1_update_ret()` function. This means the comment here is OK from an MD usage perspective, so my example is not valid. This doesn't mean there might be other examples of this use case. So, is it strictly necessary for the operation to have been started for this entry point?

Reply:
> We don't have parameter checks that can enforce this.

In general, parameter validation does not (and cannot) check all conditions mentioned in the documentation.

> It's desirable that a hash operation has started, but is it strictly required? 

I don't think so. If there's doubt about whether it's even the case in our own usage of SHA-1, I prefer to remove this requirement.